sdk/core/azure-core/inc/azure/core/http/http.hpp

```diff
@@ -603,29 +627,59 @@ namespace Azure { namespace Core { namespace Http {
     // ===== Methods used to build HTTP response =====
 
     /**
-     * @brief Add header to the HTTP response.
+     * @brief Adds an HTTP header to the RawResponse.
      *
-     * @param name HTTP response header name.
-     * @param value HTTP response header value.
+     * @remark The \p name must contain valid header name characters (RFC 2730).
+     *
+     * @param name The name for the header to be added.
+     * @param value The value for the header to be added.
+     * @param invalidHeaderIndex Optional size_t pointer to reference the first invalid character
+     * index from \p name on fail.
+     * @return true when header is successfully added.
+     * @return false when there are invalid characters on \p name and header is not added.
      */
-    void AddHeader(std::string const& name, std::string const& value);
+    bool AddHeader(
+        std::string const& name,
+        std::string const& value,
+        size_t* const invalidHeaderIndex = nullptr);
 
     /**
-     * @brief Add header to the HTTP response.
+     * @brief Adds an HTTP header to the RawResponse.
+     *
+     * @remark The \p header must contain valid header name characters (RFC 2730).
+     * @remark Header Name, value and delimiter are expected to be in \p header. If delimiter is
+     * missing, \p invalidHeaderIndex will be set to 0 and header won't be added.
      *
-     * @param header HTTP response header in RFCnnnn format (`OWS header-value OWS`).
+     * @param header The header to be added.
+     * @param invalidHeaderIndex Optional size_t pointer to reference the first invalid character
+     * index from \p header on fail.
+     * @return true when header is successfully added.
+     * @return false when there are invalid characters on \p header or delimiter is missing. Header
+     * is not added.
      */
-    // rfc form header-name: OWS header-value OWS
-    void AddHeader(std::string const& header);
+    bool AddHeader(std::string const& header, size_t* const invalidHeaderIndex = nullptr);
 
     /**
-     * @brief Add header to the HTTP response.
-     * @detail HTTP response header should be in RFCnnnn format (`OWS header-value OWS`).
+     * @brief Adds an HTTP header to the RawResponse.
```

Comment:
I am going to stop mentioning the same feedback about "Adds" => "Add", `@ClassName`, single `@return` section, Capitalizing the start of the sentence, ending with a period, backticking types (`size_t`, `std::string`, `true`, `false`, etc), marking out parameters as `@param[out]`, "Header [was not] added".